SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -93,52 +93,67 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 
 	FString ClientWorkerAttribute = GetOwnerWorkerAttribute(Actor);
 
-	WorkerAttributeSet ServerAttribute = { SpatialConstants::ServerWorkerType };
-	WorkerAttributeSet ClientAttribute = { SpatialConstants::ClientWorkerType };
-	WorkerAttributeSet OwningClientAttribute = { ClientWorkerAttribute };
+	WorkerRequirementSet AnyServerRequirementSet;
+	WorkerRequirementSet AnyServerOrClientRequirementSet = { SpatialConstants::UnrealClientAttributeSet };
 
-	WorkerRequirementSet ServersOnly = { ServerAttribute };
-	WorkerRequirementSet ClientsOnly = { ClientAttribute };
-	WorkerRequirementSet OwningClientOnly = { OwningClientAttribute };
+	WorkerAttributeSet OwningClientAttributeSet = { ClientWorkerAttribute };
+
+	WorkerRequirementSet AnyServerOrOwningClientRequirementSet = { OwningClientAttributeSet };
+	WorkerRequirementSet OwningClientOnlyRequirementSet = { OwningClientAttributeSet };
+
+	const TArray<FName>& ServerWorkerTypes = GetDefault<USpatialGDKSettings>()->WorkerTypes.Array();
+	for (const FName& ServerWorkerType : ServerWorkerTypes)
+	{
+		WorkerAttributeSet ServerWorkerAttributeSet = { ServerWorkerType.ToString() };
+
+		AnyServerRequirementSet.Add(ServerWorkerAttributeSet);
+		AnyServerOrClientRequirementSet.Add(ServerWorkerAttributeSet);
+		AnyServerOrOwningClientRequirementSet.Add(ServerWorkerAttributeSet);
+	}
 
-	WorkerRequirementSet AnyUnrealServerOrClient = { ServerAttribute, ClientAttribute };
-	WorkerRequirementSet AnyUnrealServerOrOwningClient = { ServerAttribute, OwningClientAttribute };
 
 	WorkerRequirementSet ReadAcl;
 	if (Class->HasAnySpatialClassFlags(SPATIALCLASS_ServerOnly))
 	{
-		ReadAcl = ServersOnly;
+		ReadAcl = AnyServerRequirementSet;
 	}
 	else if (Actor->IsA<APlayerController>())
 	{
-		ReadAcl = AnyUnrealServerOrOwningClient;
+		ReadAcl = AnyServerOrOwningClientRequirementSet;
 	}
 	else
 	{
-		ReadAcl = AnyUnrealServerOrClient;
+		ReadAcl = AnyServerOrClientRequirementSet;
 	}
 
 	const FClassInfo& Info = ClassInfoManager->GetOrCreateClassInfoByClass(Class);
 
+	WorkerRequirementSet AuthoritativeWorkerRequirementSet = SpatialConstants::UnrealServerPermission;
+	if (Info.WorkerType != NAME_None)
```

Comment:
Worker type is set by searching for actor group right? If there's no actor group for this actor then ActorGroupManager will fail this conditional and go below right?

```	
if (ActorGroupToWorkerType.Contains(ActorGroup))
{
	return ActorGroupToWorkerType.FindRef(ActorGroup);
}

return TEXT("");
```

Shouldn't this conditional be != TEXT("") ?

Reply:
Would you log an error and return if the workertype is empty here then or just write the code assuming it's set?

Reply:
Assume it is set here. We'll make that part of the guarantee of the ActorGroupManager api, which can deal with fatal errors.

Reply:
@mattyoung-improbable @alastairdglennie does this mean that we can remove the if check completely and just configure the WorkerAttribute to be the workertype in the `Info` variable?